sys/cpp11-compat/include/riot/condition_variable.hpp

```diff
@@ -171,20 +172,20 @@ cv_status condition_variable::wait_for(unique_lock<mutex>& lock,
   if (timeout_duration <= timeout_duration.zero()) {
     return cv_status::timeout;
   }
-  timex_t timeout, before, after;
+  uint64_t timeout, before, after;
   auto s = duration_cast<seconds>(timeout_duration);
-  timeout.seconds = s.count();
-  timeout.microseconds
-    = (duration_cast<microseconds>(timeout_duration - s)).count();
-  xtimer_now_timex(&before);
-  xtimer_t timer;
-  xtimer_set_wakeup(&timer, timex_uint64(timeout), thread_getpid());
+  timeout = s.count() * US_PER_SEC;
+  timeout += (duration_cast<microseconds>(timeout_duration - s)).count();
```

Comment:
```suggestion
  timeout = duration_cast<microseconds>(timeout_duration).count();
```
and remove line 176

Reply:
Right, I thought that microseconds overflow at 1 second but I guess that was done as a result of `timex`.  Done.